rfcs/text/0000/agent.yml

```diff
@@ -0,0 +1,7 @@
+- name: agent
+  fields:
+    - name: hostname
```

Comment:
The `agent.*` fields are meant to describe the software entity collecting events on a host or observer. As a software entity, the `agent.hostname` field has been left out intentionally since the hostname is instead an attribute of a `host.*` or an `observer.*`.

Reply:
We have seen evidence of records (observer.*) which report on a host (host.*) and regarding the agent (agent.*) where the hostnames of each (observer, host, and agent) are unique.

Reply:
Do you have an example you'd be willing to share for this discussion?

Reply:
@ebeahan 
Here is an anonymized representation of the event output from the agent (represented with 123) reporting on the HOST and the other agent deployed on the host (321). 
|application.reporting_agent.assetid | GUID123_HOSTGUID|
|--- | ---|
|application.reporting_agent.endpoint.log.level | INFO|
|application.reporting_agent.endpoint.product.version | 2.7|
|host.broker_guid | BROKERGUID|
|host.domain | HOSTNAME.DOMAIN.COM|
|host.resident_agent.name | HOSTNAME123|
|host.hostname | HOSTNAME|
|host.id | HOSTGUID|
|host.ip | HOSTIP|
|reporting_agent.server | HOSTNAME.DOMAIN.COM|
|reporting_agentguid | GUID123|
|event.resident_agent.version | 10.4|
|host.resident_agent_server.guid | GUID321|
|host.resident_agent_server.name | "HOSTNAMESVR321"|
|reporting_agent.server.name | "HOSTNAMESVR123.DOMAIN.COM"|

Reply:
Hi @hadadata59! Just catching up here and want to verify I understand correctly. 

From your example, you have agent 123 and agent 321 and both run on the same host that they are monitoring?

Reply:
@kgeller yes.  two agents, one host.

Reply:
What are thoughts on leveraging the existing `agent.name` to capture the hostname?

For example, Beats does this as the default, unless overridden in the configuration: https://github.com/elastic/beats/pull/18000

Reply:
This is a single log record (not sets of logs) where the box (host.hostname), the non-reporting 'resident' agent (?) and the reporting agent (agent.hostname) all provide unique hostnames in the record.